stats/influxdb/collector.go

```diff
@@ -133,8 +151,26 @@ func (c *Collector) commit() {
 }
 
 func (c *Collector) extractTagsToValues(tags map[string]string, values map[string]interface{}) map[string]interface{} {
-	for _, tag := range c.Config.TagsAsFields {
+tags:
+	for tag, kind := range c.fieldKinds {
 		if val, ok := tags[tag]; ok {
+			var v interface{}
+			var err error
+			switch kind {
+			case String:
+				v = val
+			case Bool:
+				v, err = strconv.ParseBool(val)
+			case Float:
+				v, err = strconv.ParseFloat(val, 64)
+			case Int:
+				v, err = strconv.ParseInt(val, 10, 64)
+			}
+			if err == nil {
+				values[tag] = v
+				delete(tags, tag)
+				continue tags
```

Comment:
I think the label isn't required

Reply:
Updated the code and removed the label and the continue statement altogether.